include/ensmallen_bits/moead/moead.hpp

```diff
@@ -0,0 +1,234 @@
+    //! Modify the mutation probability.
+    double& MutationProbability() { return mutationProb; }
+
+    //! Get the size of the weight neighbourhood.
+    size_t NeighbourhoodSize() const { return neighbourhoodSize; }
+    //! Modify the size of the weight neighbourhood.
+    size_t& NeighbourhoodSize() { return neighbourhoodSize; }
+
+    //! Retrieve value of lowerBound.
+    const arma::vec& LowerBound() const { return lowerBound; }
+    //! Modify value of lowerBound.
+    arma::vec& LowerBound() { return lowerBound; }
+
+    //! Retrieve value of upperBound.
+    const arma::vec& UpperBound() const { return upperBound; }
+    //! Modify value of upperBound.
+    arma::vec& UpperBound() { return upperBound; }
+
+    //! Retrieve the best front (the Pareto frontier).  This returns an empty
+    //! vector until `Optimize()` has been called.
+    const std::vector<arma::mat>& Front() const { return bestFront; }
+
+  private:
+    /**
+     * Mutate child formed by the crossover of two random members of the 
+     * population.
+     *
+     * @tparam MatType, the type of matrix used to store coordinates.
+     * @param child, the matrix to mutate.
+     * @param lowerBound, the lower bound on each variable in the matrix.
+     * @param upperBound, the upper bound on each variable in the matrix.
+     */
+    template<typename MatType>
+      void Mutate(MatType& child,
+                const arma::vec& lowerBound,
+                const arma::vec& upperBound);
+
+    /**
+     * Decompose the multi objective problem to a single objetive problem
+     * using the Tchebycheff approach.
+     *
+     * @param weights, a set of real number which act as weights.
+     * @param idealPoint, ideal point z in Tchebycheff decomposition.
+     * @param evaluatedCandidate, value of the candidate per objective.
+     * @return the single value obtained from decomposed function.
+     */
+    double DecomposedSingleObjective(const arma::vec& weights,
+                                     const arma::vec& idealPoint,
+                                     const arma::vec& evaluatedCandidate);
+
+    /**
+     * Check domination between two vectors.
+     *
+     * @param first, the first of the two vectors to compare.
+     * @param second, the second of the two vectors to compare.
+     * @return true if first dominates second else false.
+     */
+    bool Dominates(const arma::vec& first,
+        const arma::vec& second);
+
+    /**
+     * Evaluate objectives for the elite population.
+     *
+     * @tparam ArbitraryFunctionType std::tuple of multiple function types.
+     * @tparam MatType Type of matrix to optimize.
+     * @param population The elite population.
+     * @param objectives The set of objectives.
+     * @param calculatedObjectives Vector to store calculated objectives.
+     */    
+    template<std::size_t I = 0,
+    typename MatType,
+             typename ...ArbitraryFunctionType>
```

Comment:
Please indent the method declarations as per the [style guide](https://github.com/mlpack/mlpack/wiki/DesignGuidelines#method-declarations).